etc/init.d/hadoop_datanode

```diff
@@ -0,0 +1,276 @@
+# Short-Description: Hadoop DataNode
+# Description: The Hadoop DataNode provides block storage for the Hadoop Distributed File System (HDFS).
+### END INIT INFO
+
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Check if datanode service config exists, if yes, source it
+[ -e /etc/sysconfig/hadoop_datanode ] && . /etc/sysconfig/hadoop_datanode
+
+# Set OOM_ADJ, if defined
+OOM_ADJ=${OOM_ADJ:-0}
+
+HDFS_USER=${HDFS_USER:-hdfs}
+HADOOP_PID_DIR=${HADOOP_PID_DIR:-/var/hadoop/hdfs/run/}
+
+# Set the ulimit
+if [ -n "$NOFILE"  ]
+then
+    ulimit -n $NOFILE
+fi
+
+if [ -n "$NPROC" ]
+then
+    ulimit -u $NPROC
+fi
+
+if [[ "x" == "${JAVA_HOME}x" && -f /etc/profile.d/jdk.sh ]]
+then
+    source /etc/profile.d/jdk.sh
+fi
+
+if [ "x" == "${JAVA_HOME}x" ]
+then
+    echo "JAVA_HOME not set"
+    exit 1
+fi
+
+if [ "x" == "${HADOOP_CONF_DIR}x" ]
+then
+    source //etc/profile.d/hadoop-env.sh
+fi
+
+if [ "x" == "${HADOOP_CONF_DIR}x" ]
+then
+    echo "Hadoop environment not configured"
+    exit 1
+fi
+
+if [ "x" != "${HDFS_LOG_DIR}x" ]
+then
+    export HADOOP_LOG_DIR=$HDFS_LOG_DIR
+fi
+
+if [ "x" != "${DATANODE_OPTS}x" ]
+then
+    echo "DATANODE_OPTS is not configured"
+    exit 1
+fi
+
+
+RETVAL=0
+# USER is required by hadoop-daemon.sh
+export USER=$HDFS_USER
+SERVICE="datanode"
+SERVICENAME="DataNode"
+DESC="Hadoop ${SERVICE} daemon"
+HADOOP_PID_DIR=/var/hadoop/hdfs/run/
+
+if [ -n "$HADOOP_SECURE_DN_USER" ]; then
+  PIDFILE="${HADOOP_PID_DIR}/hadoop_secure_dn.pid"
+  LOCKFILE="${HADOOP_PID_DIR}/hadoop-root-${SERVICE}.lock"
+  PROCNAME="jsvc"
+else
+  PIDFILE="${HADOOP_PID_DIR}/hadoop-${USER}-${SERVICE}.pid"
+  LOCKFILE="${HADOOP_PID_DIR}/hadoop-${USER}-${SERVICE}.lock"
+  PROCNAME="java"
+fi
+
+# HADOOP_DATANODE_OPTS is used by ${HADOOP_HOME}/bin/hdfs
+# which is called from ${HADOOP_HOME}/sbin/hadoop-daemon.sh
+# which is called by this initialization script.
+# Note that ${HADOOP_HOME}/sbin/hadoop-daemon.sh sources
+# ${HADOOP_CONF_DIR}/hadoop-env.sh
+# bin/hdfs does not yet implement a HEAPSIZE option.
+export HADOOP_SECURITY_LOGGER="${HADOOP_SECURITY_LOGGER:-INFO,DRFAS}"
+export HADOOP_DATANODE_OPTS="$DATANODE_OPTS\
+ -Xloggc:${HDFS_LOG_DIR}/hadoop-${USER}-${SERVICE}-gc.log-`date +'%Y%m%d%H%M'`\
```

Comment:
guys, this change is 2181 lines - I don't expect it to be perfect and I don't want to change semantics of what these scripts are supposed to be doing.

So, please, keep the comments about how to refactor init.d scripts to be better for next change. I did not write these scripts and I cannot change the semantic in one PR.

The goal of this change is to refactor existing chef-solo init.d scripts into static code (in init.d/) and settings (in /etc/sysconfig/ dir). We plan to only use these new scripts (that all have new names) in chef server code, and when we launch particular service in chef server, then the old chef solo code becomes obsolete.